zebra-test/src/command.rs

```diff
@@ -3,31 +3,38 @@ use color_eyre::{
     Help, SectionExt,
 };
 use std::process::{Child, Command, ExitStatus, Output};
-use std::{fs, io::Write};
+use std::{fs, io::Write, path::PathBuf};
 use tempdir::TempDir;
 
 #[cfg(unix)]
 use std::os::unix::process::ExitStatusExt;
 
-/// Runs a command in a TempDir
-pub fn test_cmd(path: &str) -> Result<(Command, impl Drop)> {
-    let dir = TempDir::new(path)?;
-    let mut cmd = Command::new(path);
-    cmd.current_dir(dir.path());
-
-    let cache_dir = dir.path().join("state");
-    fs::create_dir(&cache_dir)?;
+/// Runs a command in already existing temp dir or create new if `tempdir_path` is `None`
+pub fn test_cmd(command_path: &str, tempdir_path: Option<&str>) -> Result<(Command, ZebraTestDir)> {
+    let dir = match tempdir_path {
+        Some(t) => ZebraTestDir::new_in(t, command_path),
+        None => {
+            let dir = ZebraTestDir::new(command_path);
+
+            // Make temp dir to be cache_dir
+            let cache_dir = dir.path().join("state");
+            fs::create_dir(&cache_dir)?;
+            fs::File::create(dir.path().join("zebrad.toml"))?.write_all(
+                format!(
+                    "[state]\ncache_dir = '{}'",
+                    cache_dir
+                        .into_os_string()
+                        .into_string()
+                        .map_err(|_| eyre!("tmp dir path cannot be encoded as UTF8"))?
+                )
+                .as_bytes(),
+            )?;
+            dir
+        }
+    };
 
-    fs::File::create(dir.path().join("zebrad.toml"))?.write_all(
-        format!(
-            "[state]\ncache_dir = '{}'",
-            cache_dir
-                .into_os_string()
-                .into_string()
-                .map_err(|_| eyre!("tmp dir path cannot be encoded as UTF8"))?
-        )
-        .as_bytes(),
-    )?;
+    let mut cmd = Command::new(command_path);
+    cmd.current_dir(dir.path());
```

Comment:
I like this approach, but rather than having the arg that we set to `None` most of the time I think it might be good to just expose two versions of `test_cmd` and `get_child`, one for new command tests and one for new commands on in progress tests. I'm thinking `get_child` and `get_child_in`.

Honestly, you don't even have to change it here, its probably fine if you only update the `get_child` fns to have the two interfaces.